acceptance logs: MkdirAll instead of Mkdir

tbg opened the pull request:
I think this is responsible for

```
F0105 17:26:51.483286 18569 acceptance/cluster/localcluster.go:369  mkdir /tmp/circle-artifacts.Z4vWa88/acceptance/TestChaos/roach0.local: permission denied
```

though there could be other issues (on OSX, `-l /tmp/something` fails, likely related to it not being mounted into the host vm).

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3625)

<!-- Reviewable:end -->

tbg committed:
acceptance logs: MkdirAll instead of Mkdir

bdarnell commented:
LGTM

tbg committed:
ci: set -x for prepare_artifacts

There is an empty excerpts.txt when it clearly should
not be empty. slow.txt was generated successfully, which
suggests that the next command failed. In general, when
something goes wrong in this section it is annoying to
debug, so -x sounds like a good idea.

tbg committed:
remove -v from curl opts

tschottdorf commented:
There's a weird bug with an empty `excerpt.txt` generated though the acceptance tests fail. I'm looking. Also seems that the commits here don't fix the original `permission denied` issue.

tbg committed:
don't use `\d` with `grep -E`

it's not supported. Use `[0-9]` instead.
(The `-P` flag supports PCREs, but OSX-grep
doesn't have that.)

tbg committed:
create log dir (or Docker will, which breaks things)

tbg committed:
CI: move acceptance before testrace

tamird commented:
LGTM

tbg committed:
add comment

tbg committed:
remove noop exists()

tamird commented:
LGTM

tschottdorf commented:
there were two greens in there somewhere and one more that ran the acceptance tests and only failed at testrace. Merging.
